pkg/ruler/compat.go

```diff
@@ -329,23 +332,23 @@ func DefaultTenantManagerFactory(
 	overrides RulesLimits,
 	reg prometheus.Registerer,
 ) ManagerFactory {
-	totalWrites := promauto.With(reg).NewCounter(prometheus.CounterOpts{
+	totalWrites := promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
 		Name: "cortex_ruler_write_requests_total",
 		Help: "Number of write requests to ingesters.",
-	})
-	failedWrites := promauto.With(reg).NewCounter(prometheus.CounterOpts{
+	}, []string{"user"})
+	failedWrites := promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
```

Comment:
don't forget to make this change in GEM's codebase too

Reply:
Oh. I didn't know there is an override. I will open a PR with the update there right after merging this one.